hw3/02-url.js

```diff
@@ -1,42 +1,47 @@
 const http = require('http');
-
-const port = process.env.PORT || 5000;
+const port = process.env.PORT || 5001;
 
 const server = http.createServer((req, res) => {
-  const routes = [
-    '/attributes?hello=world&lorem=ipsum',
-    '/items?first=1&second=2&third=3&fourth=4',
-    '/characters?spongebob=squarepants&patrick=star&sandy=cheeks',
-  ];
+    const routes = [
+        '/attributes?hello=world&lorem=ipsum',
+        '/items?first=1&second=2&third=3&fourth=4',
+        '/characters?spongebob=squarepants&patrick=star&sandy=cheeks',
+    ];
+
+    let url = new URL(req.url, `http://${req.headers.host}`);
 
-  // use the URL interface to work with URLs
-  // source: https://developer.mozilla.org/en-US/docs/Web/API/URL
-  let url = new URL(req.url, `http://${req.headers.host}`);
+    let getRoutes = () => {
+        let result = '';
 
-  let getRoutes = () => {
-    let result = '';
+        routes.forEach((elem) => (result += `<li><a href="${elem}">${elem}</a></li>`));
 
-    routes.forEach(
-      (elem) => (result += `<li><a href="${elem}">${elem}</a></li>`)
-    );
+        return result;
+    };
 
-    return result;
-  };
+    if (req.url === '/') {
+        let routeResults = getRoutes();
 
-  if (req.url === '/') {
-    let routeResults = getRoutes();
+        res.writeHead(200, { 'Content-Type': 'text/html' });
+        res.write(`<h1>Exercise 02</h1>`);
 
-    res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.write(`<h1>Exercise 02</h1>`);
+        res.write(`<ul> ${routeResults} </ul>`);
+    } else if (req.url.includes('?')) {
+        let param_string = url.toString().split('?')[1];
```

Comment:
The code on line 11 has used the URL class to build an object that has done this parsing and URLSearchParams object creation